common/tests/cmd_test.go

```diff
@@ -0,0 +1,56 @@
+package tests
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"go.k6.io/k6/cmd"
+	k6tests "go.k6.io/k6/cmd/tests"
+)
+
+func TestCmd(t *testing.T) {
+	t.Skip("data race")
+
+	t.Parallel()
+
+	ts := k6tests.NewGlobalTestState(t)
+
+	ts.CmdArgs = []string{
+		"k6", "run", "-",
+	}
+	ts.Stdin = bytes.NewBufferString(`
+		import { browser } from 'k6/browser';
+
+		export const options = {
+		scenarios: {
+			browser: {
+				executor: 'shared-iterations',
+				options: {
+				browser: {
+					type: 'chromium',
+				},
+				},
+			},
+		},
+		};
+
+		export default async function () {
+		const page = await browser.newPage();
+		await page.goto('https://test.k6.io/browser.php');
+		const options = page.locator('#numbers-options');
+		await options.selectOption({label:'Five'});
```

Comment:
@ankur22 @inancgumus any tips on how to better test this keeping in mind the current test structure?

I took a look into the `tests` package and not sure what could be the best way to emulate input like that (and below).

Reply:
Basically, I've created this to be like the [ones that k6-core uses](https://github.com/grafana/k6/blob/master/cmd/tests/eventloop_test.go#L122-L130).

My predominant concern is why it should start from JS is because partially the `panic` was caused because of some "magic" sobek type transitioning, and I'd like to keep it as part of the test, to ensure that it works

Reply:
I really like how you wrote the test, it's much more succinct to how we do it.

If we can't sort the race condition out, then take a look at a test that works with `startIteration`, such as [TestPageOnMetric](https://github.com/grafana/xk6-browser/blob/5aa78f53dad26715613ac6e83cdcc57c9c4acba2/tests/page_test.go#L1887).